### PR TITLE
feat(integration): Add Sentinel One firewall control rule + refactors

### PR DESCRIPTION
**Pull request by mattdurant**
## Description

Refactored Sentinel One integration to match suggestions for Okta integration.
Added 2 new actions:

- Get a Sentinel One Firewall Control rule
- Update a Firewall Control rule

## Related Tickets & Documents

N/A

## Screenshots/Recordings

N/A

## Steps to QA

Tested against live API.

### CODE REVIEW

File: tracecat/actions/integrations/edr/sentinel_one.py [hunk 1]
```diff
@@ -49,6 +50,21 @@
 """
 
 
+def create_sentinel_one_client() -> httpx.AsyncClient:
+    SENTINEL_ONE_API_TOKEN = os.getenv("SENTINEL_ONE_API_TOKEN")
+    if SENTINEL_ONE_API_TOKEN is None:
+        raise ValueError("SENTINEL_ONE_API_TOKEN is not set")
+    client = httpx.AsyncClient(
+        base_url=f'{os.getenv("SENTINEL_ONE_BASE_URL")}/web/api/v2.1',
```
Comment (topher-lo) [inline tracecat/actions/integrations/edr/sentinel_one.py]: @mattdurant are all S1 API endpoints globally updated under v2.1?

Comment (mattdurant) [reply tracecat/actions/integrations/edr/sentinel_one.py]: All that I'm aware of, v2.0 was missing a lot of the newer feature endpoints and there was some refactoring as well.  I think there are a lot of integrations out there for other security tools still using v2.0 and haven't been upgraded to use 2.1.

Comment (topher-lo) [reply tracecat/actions/integrations/edr/sentinel_one.py]: In that case, shall we parameterize `api_version` in `create_sentinel_one_client(api_version: str ="v2.1")` keep v2.1 as default to allow this to be reuse for any non-updated APIs?

Comment (mattdurant) [reply tracecat/actions/integrations/edr/sentinel_one.py]: It wouldn't hurt, I'm not sure I would ever use it, but I can follow that pattern when working on integrations

**Comment by topher-lo** [comment]
LGTM. Feel free to request a review if there's nothing else you want to add!

**Review by topher-lo — APPROVED** [review]

**Pull request merged**